### ISSUE TITLE
Running `otto build` on a Ruby app with package.json fails

### ISSUE THREAD

**Issue by faun** [hashicorp/otto#124]
Running `otto build` for a Ruby app results in the following error:

```
==> Detecting infrastructure credentials for: app-name (aws)
    Cached and encrypted infrastructure credentials found.
    Otto will now ask you for the password to decrypt these
    credentials.

Encrypted Credentials Password
  Infrastructure credentials are required for this operation. Otto found
  saved credentials that are password protected. Please enter the password
  to decrypt these credentials. You may also just hit <enter> and leave
  the password blank to force Otto to ask for the credentials again.

  Enter a value:
==> Querying infrastructure data for build...
==> Building deployment archive...
==> Building deployment artifact with Packer...
    Raw Packer output will begin streaming in below. Otto
    does not create this output. It is mirrored directly from
    Packer while the build is being run.

otto output will be in this color.

==> otto: Prevalidating AMI Name...
==> otto: Inspecting the source AMI...
==> otto: Creating temporary keypair: packer 560b0411-1b41-f23c-e1ba-a27ba76f169c
==> otto: Creating temporary security group for this instance...
==> otto: Authorizing access to port 22 the temporary security group...
==> otto: Launching a source AWS instance...
    otto: Instance ID: i-d1666d72
==> otto: Waiting for instance (i-d1666d72) to become ready...
==> otto: Waiting for SSH to become available...
==> otto: Connected to SSH!
==> otto: Provisioning with shell script: /var/folders/h3/y_72pc5x1mj625d652qmr9fh0000gn/T/packer-shell220436018
==> otto: Uploading /Users/user-name/src/github.com/org-name/app-name/.otto/compiled/app/foundation-consul/app-build/ => /tmp/otto/foundation-1
==> otto: Provisioning with shell script: /var/folders/h3/y_72pc5x1mj625d652qmr9fh0000gn/T/packer-shell185151415
    otto: [otto] Installing Consul...
    otto: [otto] Installing dnsmasq for Consul...
    otto: [otto] Configuring consul service: app-name
==> otto: Uploading /var/folders/h3/y_72pc5x1mj625d652qmr9fh0000gn/T/otto-slug-041027226 => /tmp/otto-app.tgz
==> otto: Provisioning with shell script: build-node.sh
    otto: [otto] Waiting for cloud-config to complete...
    otto: [otto] Adding apt repositories and updating...
    otto: [otto] Downloading Node 4.1.0...
    otto: ERROR at /tmp/script_7943.sh:13; Last logs:
    otto: Sep 29 21:38:05 ip-10-237-15-238 otto: Ign http://us-east-1.ec2.archive.ubuntu.com trusty/universe Translation-en_US
    otto: Sep 29 21:38:05 ip-10-237-15-238 otto: Ign http://security.ubuntu.com trusty-security InRelease
    otto: Sep 29 21:38:05 ip-10-237-15-238 otto: Hit http://security.ubuntu.com trusty-security Release.gpg
    otto: Sep 29 21:38:05 ip-10-237-15-238 otto: Hit http://security.ubuntu.com trusty-security Release
    otto: Sep 29 21:38:05 ip-10-237-15-238 otto: Hit http://security.ubuntu.com trusty-security/main Sources
    otto: Sep 29 21:38:05 ip-10-237-15-238 otto: Hit http://security.ubuntu.com trusty-security/universe Sources
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Hit http://security.ubuntu.com trusty-security/main amd64 Packages
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Hit http://security.ubuntu.com trusty-security/universe amd64 Packages
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Hit http://security.ubuntu.com trusty-security/main Translation-en
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Get:1 https://oss-binaries.phusionpassenger.com trusty InRelease
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Ign https://oss-binaries.phusionpassenger.com trusty InRelease
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Hit http://security.ubuntu.com trusty-security/universe Translation-en
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Get:2 https://oss-binaries.phusionpassenger.com trusty Release
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Get:3 https://oss-binaries.phusionpassenger.com trusty/main amd64 Packages
    otto: Sep 29 21:38:06 ip-10-237-15-238 otto: Get:4 https://oss-binaries.phusionpassenger.com trusty/main Translation-en_US
    otto: Sep 29 21:38:07 ip-10-237-15-238 otto: Ign https://oss-binaries.phusionpassenger.com trusty/main Translation-en_US
    otto: Sep 29 21:38:07 ip-10-237-15-238 otto: Ign https://oss-binaries.phusionpassenger.com trusty/main Translation-en
    otto: Sep 29 21:38:07 ip-10-237-15-238 otto: Fetched 53.9 kB in 2s (22.1 kB/s)
    otto: Sep 29 21:38:13 ip-10-237-15-238 otto: Reading package lists...
    otto: Sep 29 21:38:13 ip-10-237-15-238 otto: /home/vagrant/node.tar.gz: No such file or directory
==> otto: Terminating the source AWS instance...
==> otto: No AMIs to cleanup
==> otto: Deleting temporary security group...
==> otto: Deleting temporary keypair...
Build 'otto' errored: Script exited with non-zero exit status: 1
Error building app: Error executing Packer: exit status 1

The error messages from Packer are usually very informative.
Please read it carefully and fix any issues it mentions. If
the message isn't clear, please report this to the Otto project.
```

**Comment by phinze** [comment]
Hi @faun - thanks for the report.

There's a problem with the build script I'm fixing now, but it also looks like the app was improperly detected as a node app.

Can you share what `ls -la` looks like from the directory where you're running Otto?

**Comment by faun** [comment]
```
drwxr-xr-x  53 faun  staff       1802 Sep 29 15:31 .
drwxr-xr-x  10 faun  staff        340 Sep 16 11:32 ..
-rw-r--r--@  1 faun  staff       8196 Sep 23 18:46 .DS_Store
-rw-r--r--   1 faun  staff         60 Aug  1 12:07 .bowerrc
drwxr-xr-x   3 faun  staff        102 Aug 19 12:11 .bundle
-rw-r--r--   1 faun  staff        388 Aug 30 17:28 .codeclimate.yml
-rw-r--r--   1 faun  staff        889 Sep 23 17:28 .env
drwxr-xr-x  17 faun  staff        578 Sep 29 15:31 .git
-rw-r--r--   1 faun  staff        192 Sep 29 11:30 .gitignore
-rw-r--r--   1 faun  staff        198 Aug  1 12:07 .hound.yml
-rw-r--r--   1 faun  staff      36708 Aug  1 12:07 .hound_defaults.yml
-rw-r--r--   1 faun  staff        711 Sep 25 11:23 .jshintrc
drwxr-xr-x   5 faun  staff        170 Sep 29 12:56 .otto
-rw-r--r--   1 faun  staff        567 Sep 29 12:56 .ottoid
-rw-r--r--   1 faun  staff       1190 Aug 13 13:45 .overcommit.yml
-rw-r--r--   1 faun  staff          9 Aug 13 13:36 .rspec
-rw-r--r--   1 faun  staff        395 Sep 10 16:44 .rubocop.yml
-rw-r--r--   1 faun  staff          6 Sep 22 15:09 .ruby-version
-rw-r--r--   1 faun  staff        403 Aug 30 17:28 .sample.env
-rw-r--r--   1 faun  staff       1303 Aug  1 12:07 .sample.overcommit.yml
-rw-r--r--   1 faun  staff       2965 Aug  1 12:07 .scss-style.yml
-rw-r--r--   1 faun  staff        720 Aug  1 12:07 .travis.yml
-rw-r--r--   1 faun  staff         86 Aug 31 13:24 Brewfile
-rw-r--r--   1 faun  staff       3715 Aug 30 17:28 CONTRIBUTING.md
-rw-r--r--   1 faun  staff       2432 Sep 28 12:42 Gemfile
-rw-r--r--   1 faun  staff      15225 Sep 28 12:42 Gemfile.lock
-rw-r--r--   1 faun  staff        206 Sep 29 11:28 Procfile
-rw-r--r--   1 faun  staff       8778 Sep 25 11:23 README.md
-rw-r--r--   1 faun  staff        295 Aug 30 17:28 Rakefile
drwxr-xr-x  16 faun  staff        544 Sep 24 10:55 app
drwxr-xr-x  13 faun  staff        442 Sep 25 11:23 bin
-rw-r--r--   1 faun  staff        566 Sep 24 21:10 bower.json
-rw-r--r--   1 faun  staff       1005 Sep 26 08:17 circle.yml
drwxr-xr-x  14 faun  staff        476 Sep 28 12:42 config
-rw-r--r--   1 faun  staff        414 Aug  1 12:07 config.ru
drwxr-xr-x   3 faun  staff        102 Sep 24 10:55 core_ext
drwxr-xr-x   7 faun  staff        238 Sep 21 13:08 coverage
drwxr-xr-x   5 faun  staff        170 Sep 25 12:13 db
drwxr-xr-x   3 faun  staff        102 Aug  1 12:07 karma
drwxr-xr-x  18 faun  staff        612 Sep 23 11:40 lib
drwxr-xr-x   8 faun  staff        272 Sep 24 14:31 log
drwxr-xr-x  16 faun  staff        544 Sep 25 15:34 node_modules
-rw-r--r--   1 faun  staff        793 Sep 26 08:17 package.json
drwxr-xr-x   2 faun  staff         68 Sep 22 13:54 preloader.rb
drwxr-xr-x  11 faun  staff        374 Sep 25 11:23 public
drwxr-xr-x  20 faun  staff        680 Sep 25 15:32 spec
drwxr-xr-x  20 faun  staff        680 Sep 29 11:24 tmp
drwxr-xr-x   3 faun  staff        102 Aug  1 12:07 vendor
```

**Comment by phinze** [comment]
Thanks for this - the problem is the app type detection is catching the `package.json` and assuming node. Will need to make it a bit smarter.

Including an `Appfile` with ```
application {
  name = "app-name"
  type = "ruby"
}
```

Will work around the detection problem in the meantime.

**Comment by faun** [comment]
Thanks!

**Comment by nicolasbeauvais** [comment]
@phinze same thing with the [Laravel](https://github.com/laravel/laravel ) PHP framework which use a package.json file by default (and probably other frameworks..).

Maybe you just need to prioritize ruby, go and php, over nodejs in the detection process ?

**Comment by phinze** [comment]
@nicolasbeauvais ah yes you might be right - can you open a fresh issue so we can track the todo?

**Comment by nicolasbeauvais** [comment]
Oops, @justincampbell already patched it #149 ! My bad ;)

**Comment by justincampbell** [comment]
:tada:
